src/hooks/useStyleSheet.ts

```diff
@@ -1,30 +1,36 @@
+/* eslint-disable no-extra-boolean-cast */
 import { RefObject } from "react";
-
 import { useIsomorphicLayoutEffect } from "./useIsomorphicLayoutEffect";
+import { getShadowRoot } from "../utils/shadowRoot";
 import { getNonce } from "../utils/nonce";
 
 // Bundler is configured to load this as a processed minified CSS-string
 import styles from "../css/styles.css";
 
-const styleElementMap: Map<Document, HTMLStyleElement> = new Map();
+const styleElementMap: Map<Document | ShadowRoot, HTMLStyleElement> = new Map();
 
 /**
  * Injects CSS code into the document's <head>
  */
 export const useStyleSheet = (nodeRef: RefObject<HTMLDivElement>): void => {
   useIsomorphicLayoutEffect(() => {
-    const parentDocument = nodeRef.current ? nodeRef.current.ownerDocument : document;
+    const parentDocument = nodeRef.current
+      ? !!getShadowRoot(nodeRef.current)
+        ? getShadowRoot(nodeRef.current)
+        : nodeRef.current.ownerDocument
+      : document;
 
-    if (typeof parentDocument !== "undefined" && !styleElementMap.has(parentDocument)) {
-      const styleElement = parentDocument.createElement("style");
+    if (parentDocument && !styleElementMap.has(parentDocument)) {
+      const styleElement = document.createElement("style");
```

Comment:
It would be `parentDocument` because changing that to `document` would break usage with libraries like https://www.npmjs.com/package/react-frame-component.

See https://github.com/omgovich/react-colorful/pull/166/files#diff-0e642e68af9db5d41afcf32c53546956ecbe896d844a354be630f80ab312e87a